### PR TITLE
Addresses #413 crawl start UI implemented

### PR DESCRIPTION
**Pull request by nikhilrayaprolu**
Addresses #413 
Changes: a new component and a service file is added to implement the functionality of crawl.

Demo Link:   https://susper-pr-415.herokuapp.com/

Screenshots for the change: 

image
image

there might be many minor glitches that need to worked. 
This pull includes work done in cloning http://yacy.searchlab.eu/CrawlStartExpert.html maintaining the UI design of susper.com.
and functional part of the the component is almost done but not tested as a CORS error is being raised. which might require a fix on the server side to allow CORS request.

### CODE REVIEW

File: src/app/crawlstart.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    // In some advance version we can include a remote logging of errors
+    let errMsg = (error.message) ? error.message :
+      error.status ? `${error.status} - ${error.statusText}` : 'Server error';
+    console.error(errMsg); // Right now we are logging to console itself
+    return Observable.throw(errMsg);
+  }
+  startcrawljob(crawlvalues) {
+    this.http.post('http://yacy.searchlab.eu/Crawler_p.html', crawlvalues)
+      .subscribe(res => {
+
```
Comment (harshit98) [inline src/app/crawlstart.service.ts]: Why this area is empty ? `startcrawljob(crawlvalues)` the values are not being passed to the function. Also kindly maintain syntax. Use `camelCase`

**Comment by mariobehling** [comment]
>  But Crawl seems to be does not work due to CORS not enabled from server side. There are some front-end glitches as well. We'll open up them as separate issues.

Please reference new issues here.

File: src/app/crawlstart/crawlstart.component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    "crawlingDomMaxCheck": "off",
+    "crawlingDomMaxPages": 1000,
+    "crawlingQ": "off",
+    "directDocByURL": "off",
+    "storeHTCache": "off",
+    "cachePolicy": "if fresh",
+    "indexText": "on",
+    "indexMedia": "off",
+    "xsstopw": "off",
+    "collection": "user",
```
Comment (harshit98) [inline src/app/crawlstart/crawlstart.component.ts]: According to issue omit the value `xsstopw`

File: src/app/crawlstart/crawlstart.component.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "agentName": ""
+  };
+  constructor(private crawlstartservice: CrawlstartService) {
+    /*this.crawlstartservice.getcrawldefaults().subscribe(res => {
+      this.crawlvalues = res;
+    });*/
+  };
+  starcrawljob() {
```
Comment (harshit98) [inline src/app/crawlstart/crawlstart.component.ts]: `starcrawljob` => function is wrong. It should be `startcrawljob`. Also, values are not being passed. I have pointed out similar function to this. Check my other comments. : )

File: src/app/crawlstart/crawlstart.component.html [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+      <div class="form-group">
+        <div class="col-md-4"><label>Time Zone Offset info </label></div>
+        <div class="col-md-6"><input name="timezoneoffset" type="text" [(ngModel)]="crawlvalues.timezoneOffset" size="4" maxlength="4" class="form-control">
+        </div>
+      </div>
+    </div>
+  </form>
+
+  <div>
+    <button style="border-color: white;background-color: #4d90fe;border: 1px solid #3079ed" (click)="starcrawljob()"><span style="color: white">Start New Crawl Job</span></button>
```
Comment (harshit98) [inline src/app/crawlstart/crawlstart.component.html]: Bad idea to use css styles in a html file.

File: src/app/crawlstart/crawlstart.component.html [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+      </div>
+      <div class="form-group">
+        <div class="col-md-4"><label for="last update">Maximum Pages per Domain</label></div>
+        <div class="col-md-6">Use:<input type="checkbox" name="DomMaxCheck" [(ngModel)]="crawlvalues.crawlingDomMaxCheck">Page Count:
+          <input type="text" class="form-control" id="last update" name="DomMaxPages" [(ngModel)]="crawlvalues.crawlingDomMaxPages" size="6" maxlength="6">
+        </div>
+      </div>
+      <div class="form-group">
+        <div class="col-md-4">
+          <label for="site">miscelleneous contraints</label>
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change to "Miscellaneous constraints"

File: src/app/crawlstart/crawlstart.component.html [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+  <h3 style="color: #dd4b39;font-family: Arial,SansSerif">Expert Crawl Start</h3>
+  <p>
+    You can define URLs as start points for Web page crawling and start crawling here.
+    "Crawling" means that YaCy will download the given website, extract all links in it and then download the content behind these links.
+    This is repeated as long as specified under "Crawling Depth".
+    A crawl can also be started using wget and the <a href="http://www.yacy-websearch.net/wiki/index.php/Dev:APICrawler" target="_blank">post arguments</a> for this web page.
+  </p>
+  <hr>
+  <h3>Crawl Job</h3>
+  <p>A Crawl Job consist of one or more start point, crawl limitations and document freshness rules.</p>
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change to ...start "points"

File: src/app/crawlstart/crawlstart.component.html [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+  </div>
+</nav>
+
+
+
+<div class="container">
+  <h3 style="color: #dd4b39;font-family: Arial,SansSerif">Expert Crawl Start</h3>
+  <p>
+    You can define URLs as start points for Web page crawling and start crawling here.
+    "Crawling" means that YaCy will download the given website, extract all links in it and then download the content behind these links.
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change the message to:
"Crawling" implies YaCy will download the given website, extract all its links and then download the content behind these links.

This change is optional and only my suggestion, so you can ignore it if you find it unnecessary. :sweat_smile:

File: src/app/crawlstart/crawlstart.component.html [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+  <hr>
+  <h3>Clean-Up before Crawl Start</h3>
+  <form>
+    <div class="row">
+      <div class="form-group">
+        <div class="col-md-4">
+          <label>No Deletion</label>
+        </div>
+        <div class="col-md-6">
+          <input name="deleteold" [(ngModel)]="crawlvalues.deleteold" value="off" type="radio">infoDo not delete any document before the crawl is started.
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: is "info" a part of this message?

File: src/app/crawlstart/crawlstart.component.html [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+      <div class="form-group">
+        <div class="col-md-4"><label>Delete sub-path</label></div>
+        <div class="col-md-6">
+          <input name="deleteold" [(ngModel)]="crawlvalues.deleteold" value="on" type="radio">For each host in the start url list, delete all documents (in the given subpath) from that host.
+        </div>
+      </div>
+      <div class="form-group">
+        <div class="col-md-4"><label>Delete only old</label></div>
+        <div class="col-md-6">
+          <input name="deleteold" [(ngModel)]="crawlvalues.deleteold" value="age" type="radio">Treat documents that are loaded ago as stale and delete them before the crawl is started.
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change "are loaded ago"  to "were loaded before"

File: src/app/crawlstart/crawlstart.component.html [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+      <div class="form-group">
+        <div class="col-md-4"><label>No Doubles</label></div>
+        <div class="col-md-6">
+          <input name="recrawl" [(ngModel)]="crawlvalues.recrawl" value="nodoubles" type="radio">Never load any page that is already known. Only the start-url may be loaded again
+        </div>
+      </div>
+      <div class="form-group">
+        <div class="col-md-4"><label>Reload</label></div>
+        <div class="col-md-6">
+          <input name="recrawl" [(ngModel)]="crawlvalues.recrawl" value="nodobles" type="radio">Treat documents that are loaded ago as stale and load them again. If they are younger, they are ignored.
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change "are loaded ago" to "were loaded before"

File: src/app/crawlstart/crawlstart.component.html [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+    <div class="row">
+      <div class="form-group">
+        <div class="col-md-4"><label>Max Depth for Snapshots</label></div>
+        <div class="col-md-6">
+          <input type="text" name="snapshotsmaxdepth" class="form-control" [(ngModel)]="crawlvalues.snapshotsMaxDepth" size="2" maxlength="2" value="-1">
+        </div>
+      </div>
+      <div class="form-group">
+        <div class="col-md-4"><label>Multiple Snapshot Versions</label></div>
+        <div class="col-md-6"><input name="replaceold" [(ngModel)]="crawlvalues.snapshotsReplaceOld" value="on" type="radio">replace old snapshots with new one
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change to ..with "new ones"

File: src/app/crawlstart/crawlstart.component.html [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+        <div class="col-md-6"><input name="replaceold" [(ngModel)]="crawlvalues.snapshotsReplaceOld" value="on" type="radio">replace old snapshots with new one
```
Comment (Marauderer97) [inline src/app/crawlstart/crawlstart.component.html]: Please change to ..with "new ones"

**Comment by Marauderer97** [comment]
Great work, :smile: A few suggestions/doubts though:
 1. Is there an extra input box here, 
image
Why does only one of them have a default value? Also is user the intended default value?
2. The text from the radio button in the second option has overflown
image

File: src/app/crawlstart.service.ts (hunk 1 above)
Comment (nikhilrayaprolu) [inline src/app/crawlstart.service.ts]: right now we aren't receiving any response from the server due to unauthorised status, once we get some response we could decide upon what to do after getting a response.

File: src/app/crawlstart/crawlstart.component.ts (hunk 2 above)
Comment (nikhilrayaprolu) [inline src/app/crawlstart/crawlstart.component.ts]: we should actually shouldn't be directly referencing the default values here, we have to receive them from server side but right now http://yacygrid.com:8300/yacy/grid/crawler/defaultValues.json?callback=p is down.

File: src/app/crawlstart/crawlstart.component.ts (hunk 3 above)
Comment (nikhilrayaprolu) [inline src/app/crawlstart/crawlstart.component.ts]: no look at the body of the function we don't need to pass any parameters to this function.

**Comment by nikhilrayaprolu** [comment]
@Orbiter addition of authentication step is done. but since a jsonp request expects a response in jsonp but the response from crawler_p.html is a html file.Due to which the client couldn't parse the response.can something could be done such that there will be a jsonp response from the server side?
image

**Comment by Orbiter** [comment]
I have added a json endpoint which you can call with the jsonp property i.e. "callback=p".
Please use Crawler_p.json now as endpoint instead of Crawler_p.html and it thus will be possible to end-to-end this.

**Comment by nikhilrayaprolu** [comment]
Receiving success callback from the server using the new jsonp endpoint
image

**Comment by nikhilrayaprolu** [comment]
@harshit98 @Marauderer97 please review.

**Comment by harshit98** [comment]
@nikhilrayaprolu Looks great! :100:

**Pull request merged**